### ISSUE TITLE
no-static-element-interactions rule do not work well

### ISSUE THREAD

**Issue by lyz810** [jsx-eslint/eslint-plugin-jsx-a11y#850]
```jsx
render() {
const { multiple } = this.props;
const role = multiple ? 'checkbox' : 'radio';
  return <div role={role} onClick={this.handleClick} onKeyDown={this.handleClick}>xxx</div>
}
```
got `Static HTML elements with event handlers require a role.`, when `role` is a variable

**Comment by ljharb** [comment]
This is just the limits of static analysis. The lint rule could be made more complex to handle this specific case, but it wouldn't handle most other dynamic cases - instead, you should define the role statically:
```jsx
if (multiple) {
  return <div role="multiple" onClick={this.handleClick} onKeyDown={this.handleClick}>xxx</div>;
}
return <div role="radio" onClick={this.handleClick} onKeyDown={this.handleClick}>xxx</div>;
```

**Comment by lyz810** [comment]
> This is just the limits of static analysis. The lint rule could be made more complex to handle this specific case, but it wouldn't handle most other dynamic cases - instead, you should define the role statically:
> 
> ```js
> if (multiple) {
>   return <div role="multiple" onClick={this.handleClick} onKeyDown={this.handleClick}>xxx</div>;
> }
> return <div role="radio" onClick={this.handleClick} onKeyDown={this.handleClick}>xxx</div>;
> ```

Maybe we can check this rule by following steps:
1. if no `rule` attribute given, check fail
2. if `rule` attribute is an empty string or any other string value you think is not correct ,check fail
3. if `rule` attribute is any literal you know, check fail
4. if `rule` attrbute is given but is not in above cases, check pass

In a word, `rule` attribute is given, and you don't know if it is not correct, then consider it as correct (it can be enabled by {strict: fasle} config , or default enabled)

**Comment by ljharb** [comment]
That would let things pass that shouldn't, which is far worse than the current scenario of letting things fail that shouldn't.

If you aren't willing to write your code in a less dynamic fashion, then your only remaining option is to override or disable the rule.

**Comment by lyz810** [comment]
> 



> That would let things pass that shouldn't, which is far worse than the current scenario of letting things fail that shouldn't.
> 
> If you aren't willing to write your code in a less dynamic fashion, then your only remaining option is to override or disable the rule.

I find the solution, in source code:
```
'jsx-a11y/no-static-element-interactions': ['error', {
      allowExpressionValues: true,
    }],
```
but it's not in docs, you can add this config into markdown

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/no-static-element-interactions.js#L62

**Comment by ljharb** [comment]
Fair enough, i didn’t realize we already supported this. A PR to add it to the docs is appreciated.

I’ll close this as answered.
